document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java

```diff
@@ -0,0 +1,120 @@
+package org.hypertrace.core.documentstore.postgres;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.typesafe.config.Config;
+import org.hypertrace.core.documentstore.Collection;
+import org.hypertrace.core.documentstore.Datastore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.*;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class PostgresDatastore implements Datastore {
+  
+  private static final Logger LOGGER = LoggerFactory.getLogger(PostgresDatastore.class);
+  
+  private static final String DEFAULT_DB_NAME = "default_db";
```

Comment:
I wonder if this is actually needed. It looks a bit arbitrary to me (unless I am missing something). In the docs I could only find the `postgres` as the only default DB: https://www.postgresql.org/docs/9.1/app-initdb.html

Reply:
I see this is to keep parity with https://github.com/hypertrace/document-store/blob/main/document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoDatastore.java#L18 but do we want to repeat that, is there any true use case where there is a `default_db` somewhere ready to be used? cc @adriancole @aaron-steinfeld

Reply:
I don't have the context on why this was there (just made my first commit to doc store this week!) but I'd agree with what I think you're saying - we shouldn't make any assumptions about the database structure in the code, and if for some reason we need to, it better match the default installation of the implementation database. It looks like here we take in an alternative via config, which is better than the mongo implementation.

Reply:
This was just kept for parity in reference to MongoDB, Postgres supports default connection, and I agree with you on that. Will make the change.

Reply:
I am sorry I did not explain myself correctly when I pointed out postgres documentation. My thoughts were that we should not make any assumption on users defaults and that the only default database I heard about was this `postgres` DB. I would suggest that we do not use any default database name and instead we require it from user as even if postgres provides one, user might decide to delete it. Instead if we ask the user to pass one database name, they should take care of that.

Reply:
Currenlty, I have added the functionality where user can pass the database from config. In case the user doesn't pass any database name, then it fallbacks to Postgres's default database i.e. `postgres`.

Reply:
As I said we don't necessarily have to repeat what we did in the past related to this, I'd rather not have a default value and require it from user cc @kotharironak @buchi-busireddy

Reply:
Have removed the default value for Postgres, should I add an exception in case the user doesn't provide database name?

Reply:
drive-by 2p is that in both cases, the default DB should be overridable and "hypertrace" as it is inevitable that people will want to re-use an existing DB and not necessarily overlap. Plus when you do integration tests, it is helpful to be accustomed to never using a default db or something confusingly named default db which isn't default.

Reply:
ex in zipkin in integration tests we frequently vary the DB, of course it depends on each test gets a new container or not, but at least in logs it is still helpful sometimes.